azure-pipelines-regression-test.yml

```diff
@@ -0,0 +1,309 @@
+      displayName: Build F# compiler and create packages
+
+    - task: PublishPipelineArtifact@1
+      displayName: Publish F# Compiler Artifacts
+      inputs:
+        targetPath: '$(Build.SourcesDirectory)/artifacts'
+        artifactName: 'FSharpCompilerArtifacts'
+        publishLocation: pipeline
+      condition: succeeded()
+
+    - task: PublishPipelineArtifact@1
+      displayName: Publish UseLocalCompiler props file
+      inputs:
+        targetPath: '$(Build.SourcesDirectory)/UseLocalCompiler.Directory.Build.props'
+        artifactName: 'UseLocalCompilerProps'
+        publishLocation: pipeline
+      condition: succeeded()
+
+  # Test against third-party repositories
+  - ${{ each item in parameters.testMatrix }}:
+    - job: Test_${{ replace(item.repo, '/', '_') }}
+      displayName: 'Regression Test: ${{ item.displayName }}'
+      dependsOn: BuildCompiler
+      pool:
+        name: $(DncEngPublicBuildPool)
+        demands: ImageOverride -equals windows.vs2022.amd64.open
+      timeoutInMinutes: 60
+      variables:
+        TestRepoName: ${{ item.repo }}
+        TestCommit: ${{ item.commit }}
+        BuildScript: ${{ item.buildScript }}
+        DisplayName: ${{ item.displayName }}
+      steps:
+      - checkout: none
+        displayName: Skip default checkout
+
+      # Download the F# compiler artifacts
+      - task: DownloadPipelineArtifact@2
+        displayName: Download F# Compiler Artifacts
+        inputs:
+          artifactName: 'FSharpCompilerArtifacts'
+          downloadPath: '$(Pipeline.Workspace)/FSharpCompiler'
+
+      - task: DownloadPipelineArtifact@2
+        displayName: Download UseLocalCompiler props
+        inputs:
+          artifactName: 'UseLocalCompilerProps'
+          downloadPath: '$(Pipeline.Workspace)/Props'
+
+      # Checkout the third-party repository at specific commit
+      - script: |
+          echo Cloning repository: $(TestRepoName)
+          git clone https://github.com/$(TestRepoName).git $(Pipeline.Workspace)/TestRepo
+          cd $(Pipeline.Workspace)/TestRepo
+          
+          echo Checking out commit: $(TestCommit)
+          git checkout $(TestCommit)
+          
+          echo Successfully checked out $(TestRepoName) at commit $(TestCommit)
+          git log -1 --oneline
+          
+          echo Repository structure:
+          dir /B
+          
+          echo Verifying build script exists: $(BuildScript)
+          if exist "$(BuildScript)" (
+            echo ✓ Build script found: $(BuildScript)
+          ) else (
+            echo ✗ Build script not found: $(BuildScript)
+            echo Available files in root:
+            dir
+            exit /b 1
+          )
+        displayName: 'Checkout $(DisplayName) at specific commit'
+
+      # Inject UseLocalCompiler.Directory.Build.props
+      - script: |
+          cd $(Pipeline.Workspace)/TestRepo
+          echo Creating Directory.Build.props to use local F# compiler
```

Comment:
This is wrong/oncomplete.
You have to detect if Directory.Build.props exists - most of the projects will have it!
If they do have it, you should include the <Import> into it.

Might not be appropriate to use cmd for xml manipulation...
Btw. if the logic gets bigger, it might make sense to extract the logic into a separate file, e.g. a `PrepareRepoForTesting.fsx` F# script file, and then `dotnet fsi PrepareRepoForTesting.fsx` it.